app/models/miq_schedule.rb

```diff
@@ -428,4 +428,10 @@ def v_zone_name
     return "" if zone.nil?
     zone.name
   end
+
+  def self.find_by_report_id(report_id)
```

Comment:
On second thought, I wonder if this would have a better home in `MiqReport`, since you have the `add_schedule` method there?